controller/eventController.js

```diff
@@ -0,0 +1,223 @@
+                        status: StatusEnum['ERROR'],
+                        message: 'Internal server error'
+                    });
+                })
+            }
+        });
+    } else { //Case where the user is not authorized
+        return res.status(HttpStatus.UNAUTHORIZED).json({
+            status: StatusEnum['ERROR'],
+            message: 'Unauthorized access.'
+        });
+    }
+
+
+};
+
+const getById = (req, res) => {
+    const eventId = req.params.id;
+    const user = req.decoded;
+
+    if (!eventId || !user) {
+        return res.status(HttpStatus.BAD_REQUEST).json({
+            status: StatusEnum['ERROR'],
+            message: 'Bad request'
+        });
+    }
+
+    Event.findOne({
+        _id: new ObjectId(eventId),
+        companyID: new ObjectId(user.companyID)
+    }).then(event => {
+        if (event) {
+            //If the user is admin then we can send him the event
+            if (ValidationUtil.isUserAdmin(user.accessType)) {
+                return res.status(HttpStatus.OK).json(event);
+            //If the user is regular, we need to check for the invite
+            } else {
+                Invite.findOne({ userID: new ObjectId(user._id), eventID: new ObjectId(event._id) })
+                    .then(invite => {
+                        if (invite) { //If the invite exists we return the event
+                            return res.status(HttpStatus.OK).json(event);
+                        }else{
+                            return res.status(HttpStatus.NOT_FOUND).json({
+                                status: StatusEnum['ERROR'],
+                                message: 'Event not found.'
+                            });
+                        }
+                    }).catch(err => {
+                        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+                            status: StatusEnum['ERROR'],
+                            message: 'Internal server error.'
+                        });
+                    });
+            }
+        } else {
+            return res.status(HttpStatus.NOT_FOUND).json({
+                status: StatusEnum['ERROR'],
+                message: 'Event not found.'
+            });
+        }
+    }).catch(err => {
+        return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json({
+            status: StatusEnum['ERROR'],
+            message: 'Internal server error.'
+        });
+    });
+};
+
+const update = (req, res) => {
+    const eventId = req.params.id;
+    const user = req.decoded;
+
+    //Only admins can update event objects
+    if (!user || !ValidationUtil.isUserAdmin(user.accessType)) {
+        return res.status(HttpStatus.UNAUTHORIZED).json({
+            status: StatusEnum['ERROR'],
+            message: 'Unauhorized.'
+        })
+    }
+
+    //TO DISCUSS
+    //When updating a field that previusly didn't exist, update works 
+    //but response returns an object without the new field
+    //how to solve it?
```

Comment:
Look in `companyController` or `eventQuestionController`. There are update methods with an `options` object. That `options` object tells `mongoose` to use validation on update parameters, to return the new object and similar.

Reply:
Will look into it.